src/Rules/Methods/FriendMethodCallRule.php

```diff
@@ -0,0 +1,93 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Methods;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\PhpDoc\Tag\FriendTag;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\FileTypeMapper;
+use PHPStan\Type\TypeWithClassName;
+
+/**
+ * @implements \PHPStan\Rules\Rule<\PhpParser\Node\Expr\MethodCall>
+ */
+class FriendMethodCallRule implements \PHPStan\Rules\Rule
+{
+
+	private FileTypeMapper $fileTypeMapper;
+
+	public function __construct(FileTypeMapper $fileTypeMapper)
+	{
+		$this->fileTypeMapper = $fileTypeMapper;
+	}
+
+	public function getNodeType(): string
+	{
+		return MethodCall::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!$node->name instanceof Node\Identifier) {
+			return [];
+		}
+		$methodName = $node->name->name;
+		$callee = $scope->getType($node->var);
+		if (!$callee->hasMethod($methodName)->yes()) {
+			return [];
+		}
+		$method = $callee->getMethod($methodName, $scope);
+		if (!$scope->canCallMethod($method)) {
+			return [];
+		}
+		$docComment = $method->getDocComment();
+		if ($docComment === null) {
+			return [];
+		}
+		$friends = $this->fileTypeMapper->getResolvedPhpDoc(
+			$method->getDeclaringClass()->getFileName(),
```

Comment:
I'm not sure how to approach that this can be `false`. Should it also result in an early exit?

Reply:
Yes, it means you're calling some internal PHP method.

Reply:
Makes sense :)